pkg/security/security_profile/activity_tree/process_node.go

```diff
@@ -149,31 +149,39 @@ func (pn *ProcessNode) scrubAndReleaseArgsEnvs(resolver *sprocess.Resolver) {
 }
 
 // Matches return true if the process fields used to generate the dump are identical with the provided model.Process
-func (pn *ProcessNode) Matches(entry *model.Process, matchArgs bool) bool {
-	if pn.Process.FileEvent.PathnameStr == entry.FileEvent.PathnameStr {
-		if sprocess.IsBusybox(entry.FileEvent.PathnameStr) {
-			panArg0, _ := sprocess.GetProcessArgv0(&pn.Process)
-			entryArg0, _ := sprocess.GetProcessArgv0(entry)
-			if panArg0 != entryArg0 {
-				return false
-			}
+func (pn *ProcessNode) Matches(entry *model.Process, matchArgs bool, normalize bool) bool {
+	if normalize {
+		// should convert /var/run/1234/runc.pid + /var/run/54321/runc.pic into /var/run/*/runc.pid
+		match, pattern := utils.PathPatternBuilder(pn.Process.FileEvent.PathnameStr, entry.FileEvent.PathnameStr, utils.PathPatternBuilderOpts{WildcardLimit: 3, PrefixNodeRequired: 1, SuffixNodeRequired: 1, NodeSizeLimit: 8})
+		if !match {
+			return false
```

Comment:
Is this expected to return `false` early here? Shouldn't we carry-on and perform the busybox and args checks as well?

Reply:
I think it is OK as the busybox check is for busybox with symlinks meaning that the path should be equal before checking the args ?

Reply:
You're right, I forgot the two busybox processes will have the same path